wfe2/wfe.go

```diff
@@ -854,6 +854,15 @@ func (wfe *WebFrontEndImpl) prepAuthorizationForDisplay(request *http.Request, a
 	}
 	authz.ID = ""
 	authz.RegistrationID = 0
+
+	// The ACME spec forbids allowing "*" in authorization identifiers. Boulder
+	// allows this internally as a means of tracking when an authorization
+	// corresponds to a wildcard request (e.g. to handle CAA properly). We strip
+	// the "*." prefix from the Authz's Identifier's Value here to respect the law
+	// of the protocol.
+	if strings.HasPrefix(authz.Identifier.Value, "*.") {
+		authz.Identifier.Value = strings.TrimPrefix(authz.Identifier.Value, "*.")
```

Comment:
I think I brought this up on the last PR but is there something we could do to imply to the user that this is an authorization for a wildcard identifier, this seems like it'll cause confusion down the road. (Perhaps this is a case where our implementation could lead the standardization of something in the spec instead of vice versa)

Reply:
I'm open to suggestions on what to do here. The simplest idea I have is that we could add a bool "Wildcard" field to the authorization. ¯\\\_(ツ)\_/¯.

Reply:
Yeah I think that work.

Reply:
Ok! I implemented this in https://github.com/letsencrypt/boulder/pull/3252/commits/2f08e06be3863ff965827833b21b13c7fd0fa178 and updated the corresponding WFE2 test.